app/scripts/metamask-controller.js

```diff
@@ -6674,6 +6675,18 @@ export default class MetamaskController extends EventEmitter {
           );
         return chainId;
       },
+      setTokenNetworkFilter: (chainId) => {
+        const { tokenNetworkFilter } =
+          this.preferencesController.getPreferences();
+        if (chainId && Object.keys(tokenNetworkFilter).length === 1) {
+          this.preferencesController.setPreference('tokenNetworkFilter', {
+            [chainId]: true,
+          });
+        }
+      },
```

Comment:
this exists in two places in MetaMaskController now. not sure if worth DRYing or not